[FIX] resource_booking: allow combination with other kind of leaves

yajo opened the pull request:
When `_leave_intervals_batch()` in `resource.calendar` is extended by other modules that add more leaves for any reason, the returned recordset must be of the `resource.calendar.leaves` model. Otherwise, Odoo fails with:

    TypeError: '<' not supported between instances of 'resource.resource' and 'resource.calendar.leaves'

The resource didn't have a meaning here, so changing it to avoid bug.

@Tecnativa TT31249 TT32190 - backport of https://github.com/OCA/calendar/pull/38

pedrobaeza commented:
Isn't this a problem of `hr_holidays_public` instead? We fixed a similar thing in 13.0 cc @victoralmau

pedrobaeza reviewed (approved):
I see that the problem is here, but the solution was the same as we did in `hr_holidays_public`

pedrobaeza commented:
/ocabot merge patch

pedrobaeza commented:
Please check Travis. Probably the test was not correct (although the fix is). Didn't you run it locally?

yajo commented:
It was my fault on a conflict resolution... I introduced a syntax error. Fixed now.

/ocabot merge patch

yajo commented:
Ah wait, there's also a bug in the test as you said. I cancelled the merge.

Commit:
[FIX] resource_booking: allow combination with other kind of leaves

When `_leave_intervals_batch()` in `resource.calendar` is extended by other modules that add more leaves for any reason, the returned recordset must be of the `resource.calendar.leaves` model. Otherwise, Odoo fails with:

    TypeError: '<' not supported between instances of 'resource.resource' and 'resource.calendar.leaves'

The resource didn't have a meaning here, so changing it to avoid bug.

@Tecnativa TT31249

yajo commented:
Finally, the good one!

/ocabot merge patch
